### PR TITLE
Implement verbose logging flag for all commands. Wrap `t2 provision` error in verbose mode. Fixes gh-216

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

**Comment by rwaldron** [comment]
This may be partly addressed in https://github.com/tessel/t2-cli/pull/215, so let's wait until that lands and I will see where this fits after that.

File: Gruntfile.js [hunk 1]
```diff
@@ -87,4 +87,16 @@ module.exports = function(grunt) {
   // Default task.
   grunt.registerTask('default', ['test']);
 
+
+  // Support running a single test suite
+  grunt.registerTask('nodeunit:only', 'Run a single test specified by a target; usage: "grunt nodeunit:only:<module-name>[.js]"', function(file) {
```
Comment (johnnyman727) [inline Gruntfile.js]: Oh thank god. I was just editing my grunt file to only run one file with `grunt nodeunit`. This is much better.

File: bin/tessel-2.js [hunk 2]
```diff
@@ -172,13 +210,21 @@ parser.command('key')
...
   })
   .callback(function(opts) {
+    logs.verbose = opts.verbose;
     key(opts)
       .then(function() {
         logs.info('Key successfully generated.');
       })
       .then(closeSuccessfulCommand, closeFailedCommand);
+  })
+  .option('verbose', {
```
Comment (johnnyman727) [inline bin/tessel-2.js]: It seems like many commands use these same settings. Could you refactor the 
`{
   default: false,
   flag: true,
   abbr: 'v',
    help: verbose
}`
object into it's own variable and assign it to each `'verbose'` option?

**Comment by johnnyman727** [comment]
@rwaldron Looks good. I only have the one suggestion about the options var.

How do you feel like the verbose option usage differs from passing `DEBUG=` to the command? It seems like we might want to centralize on one debugging method. What do you think?

File: bin/tessel-2.js (hunk 2 above)
Comment (rwaldron) [inline bin/tessel-2.js]: Derp... why didn't I just do that to begin with??

Comment (johnnyman727) [inline bin/tessel-2.js]: :)

**Comment by rwaldron** [comment]
Updated. 


> How do you feel like the verbose option usage differs from passing DEBUG= to the command? It seems like we might want to centralize on one debugging method. What do you think?

I'm not sure actually... Let me think about that a bit more. In the meantime, I can move the grunt task to another PR

**Comment by Frijol** [comment]
Any update on this?

**Comment by johnnyman727** [comment]
@rwaldron should we close this PR? The one thing I liked about it was the ability to specify a nodeunit file to run.

**Comment by johnnyman727** [comment]
@rwaldron closing this for now since we already have the Grunt single file tests in master. Feel free to re-open if you think this PR is relevant.

**Pull request closed without merging**